### PR TITLE
Added document on using resources.

### LINKED ISSUE #9753: api documentation is not aimed at kubectl users
We have lots of api documentation in docs/api.md and docs/api-conventions.md.  But it is focused
on direct REST access and on how to extend the API.

We need documentation which is for people who:
  - will interact mostly with kubectl 
    - gloss over REST/HTTP
  - don't plan on extending the API 
    - less prescriptive tone, less design discussion
  - are not thinking about the problem of multiple api versions yet
   - deemphasize the version stuff.

### PR DESCRIPTION
**Pull request by erictune**
Fixes #9753

### CODE REVIEW

File: docs/working_with_resources.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+pods/original
+$ wc -l original.yaml current.yaml
+      51 current.yaml
+       9 original.yaml
+      60 total
+```
+The resource we posted had only 9 lines, but the one we got back had 51 lines. 
+If you `diff original.yaml current.yaml`, you can see the fields added to the pod.
+The system adds fields in several ways:
+  - Some fields are added synchronously with creation of the resource and some are set synchronously.
```
Comment (derekwaynecarr) [inline docs/working_with_resources.md]: Asynchronously

**Comment by erictune** [comment]
fixed

File: docs/working_with_resources.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Working with Resources
+
+*This document is aimed at users who have worked through some of the examples,
+and who want to learn more about using kubectl to manage resources such
+as pods and services.  Users who want to access the REST API directly,
+and developers who want to extend the kubernetes API should 
+refer to the [api conventions](../docs/working_with_resources.md) and
```
Comment (mikedanese) [inline docs/working_with_resources.md]: api conventions links to this document. It should link to ../docs/api-conventions.md.

File: docs/working_with_resources.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+$ wc -l original.yaml current.yaml
+      51 current.yaml
+       9 original.yaml
+      60 total
+```
+The resource we posted had only 9 lines, but the one we got back had 51 lines. 
+If you `diff original.yaml current.yaml`, you can see the fields added to the pod.
+The system adds fields in several ways:
+  - Some fields are added synchronously with creation of the resource and some are set asynchronously.
+    - For example: `metadata.uid` is set synchronously.  (Read more about [metadata](../docs/api-conventions.md#metadata).
```
Comment (mikedanese) [inline docs/working_with_resources.md]: first parenthesis is untermintated. need one after last period.

File: docs/working_with_resources.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    - For example, `status.hostIP` is set only after the pod has been scheduled.  This often happens fast, but you may notice pods which do not have this set yet.  This is called Late Initialization.  (Read mode about [status](../docs/api-conventions.md#specandstatus) and [late initialization](../docs/api-conventions.md#lateinit) ).
+  - Some fields are set to default values.  Some defaults vary by cluster and some are fixed for the API at a certain version.  (Read more about [defaulting](../docs/api-conventions.md#defaulting).
```
Comment (mikedanese) [inline docs/working_with_resources.md]: first parenthesis on this line is untermintated.

**Comment by mikedanese** [comment]
A blurb about immutable fields might be appropriate to add to `docs/working_with_resources.md` but that can be added in a follow up.

File: docs/working_with_resources.md (hunk 2 above)
Comment (erictune) [inline docs/working_with_resources.md]: fixed in next push.

File: docs/working_with_resources.md (hunk 4 above)
Comment (erictune) [inline docs/working_with_resources.md]: fixed

File: docs/working_with_resources.md (hunk 3 above)
Comment (erictune) [inline docs/working_with_resources.md]: fixed.

**Pull request merged**